### ISSUE TITLE
*** No instances to display *** 

### ISSUE THREAD

**Issue by zlm-hhu** [matterport/Mask_RCNN#2886]
I am a  Rookie who just start to learn mask R-CNN recently. After installing all the dependencies , I run the 'demo' code in Jupyter. The code proceed successfully without error, but there is no segmentation on the picture as expected. And the output of the result says "*** No instances to display *** ". Could anyone help me with that? Thanks!
best wishes
LmZheng

**Comment by JPVercosa** [comment]
I'm going through the same problem.

I'm using [this branch from akTwelve](https://github.com/akTwelve/Mask_RCNN), that updated the model to TensorFlow>=2.0

I'm training in a custom dataset, and mask-rcnn was working fine until last week. As I'm working with 200+ classes my precision wasn't good, but the model was at least trying to recognise objects when using Inference config.

I've trained a new version of the model with more epochs than previous one, and it semmed to worked. But every picture I tried always show "*** No instances to display *** " when using `visualize.display_instances()`.

I don't know if it's something in the model/packages that isn't working properly, but I don't think it was something I changed it in my code.

I will try some fixes. If you can solve it, please let me know.

If someone can help us. I was parametrizing the model and training different versions as I saw that display_instances() was not showing instances anymore, cause the model can't locate then.

What worries me is that I made it work more than once and from one week to the next the model seemed to stop working. Even when I go back to old versions of .h5 files and load the weights of those old versions the Mask-RCNN model doesn't seem to be able to locate the objects anymore.

**Comment by JPVercosa** [comment]
Taking a look at issue [#2834](https://github.com/matterport/Mask_RCNN/issues/2834), people said that the problem is probably with h5py or keras version. 

I think I just fixed the problem running `pip install -r requirements.txt` on my Jupyter enviroment. 
I'm on Windows with TensorFlow 2.5.0 and CUDA 11.2 and cuDNN 8.1.

**Comment by zlm-hhu** [comment]
Thank you for your advice. But it still doesn't work. Firstly, I tried to reinstall the h5py in different versions in my original environment (Tensorflow-gpu1.11.0, Keras 2.2.4, CUDA 9, and cuDNN 7.6.5) , but the result still remained unchanged (proceed successfully without error, but no segmentation on the picture as expected). Secondly, I built a new environment based on yours (Tensorflow-gpu 2.5.0, Keras 2.10.0, CUDA 11.2 and cuDNN 8.1), the program complained with the error "module 'keras.engine' has no attribute 'Layer'". I am wondering if it is due to the Python or Keras version? Which Python and Keras version do you use in your environment? Or maybe there is another reason? 
Anyway, thank you very much.

**Comment by JPVercosa** [comment]
Hey, here you can see a list of some of my packages on the enviroment where the code is currently working and their versions. 

```
h5py                          3.1.0
ipykernel                     6.15.2
keras-nightly                 2.5.0.dev2021032900
Keras-Preprocessing           1.1.2
mask-rcnn                     2.1
matplotlib                    3.5.3
numpy                         1.19.5
pandas                        1.4.4
scikit-image                  0.16.2
tensorflow                    2.5.0
tensorflow-estimator          2.5.0
tensorflow-gpu                2.9.1
```

I'm using Python 3.8, but I dont think that should be the problem. I've have faced this error before "module 'keras.engine' has no attribute 'Layer'", but it worked with Tensorflow-2.5.0. 

I had to fork/clone [this repository from akTwelve](https://github.com/akTwelve/Mask_RCNN) to work with the above enviroment specs. Is the same repo structure and files from matterport mask-rcnn implementation, but the model is updated to  TensorFlow>2.0.

I dont think you have to install a specific version of Keras, because when you install the correct version of TensorFlow it will came with the necessary Keras.

You said that you are trying to run the demo code, check this snippet on your machine and see if the model is correctly loading the weights from COCO:

```
# Create model object in inference mode.
model = modellib.MaskRCNN(mode="inference", model_dir=MODEL_DIR, config=config)

# Load weights trained on MS-COCO
model.load_weights(COCO_MODEL_PATH, by_name=True)
```

Remember that when you are working with coco Dataset, as seen in the akTwelve branch [(here)](https://github.com/akTwelve/Mask_RCNN), you need to install some extra stuff:

![Requirements for COCO](https://user-images.githubusercontent.com/27090431/192623707-155306b0-b493-47a7-bb8c-5e0e51181ce6.png)

I hope you can solve it!

**Comment by zlm-hhu** [comment]
WOW! It finally worked after I tried the code based on tensorflow > 2.0 although it still confuesd me why the code did not work in my original environment. 

Really appreciate your help!

**Comment by parthkvv** [comment]
> WOW! It finally worked after I tried the code based on tensorflow > 2.0 although it still confuesd me why the code did not work in my original environment.
> 
> Really appreciate your help!

Can you mention your keras and tensorflow packages along with their versions? I am stuck on a similar error, and any help would be appreciated.
Thank you.

**Comment by zlm-hhu** [comment]
> 

Tensorflow-gpu2.5.0+python3.8.0+CUDA11.2+cuDNN8.1.0. You don't have to install keras individually because since tf2.0, keras has been intergrated into the tf package. Remember if you try to use this env, you have to  use the code based on tensorflow > 2.0 which could been downloaded from here (https://github.com/matterport/Mask_RCNN).

**Comment by JPVercosa** [comment]
Actually if you want to use tensorflow > 2.0, the updated code for Mask-RCNN is on this repository https://github.com/akTwelve/Mask_RCNN.

@zlm-hhu i think you can close this issue, cause you have already solve it

**Comment by parthkvv** [comment]
Thank you for the help @zlm-hhu @JPVercosa.
